src/olgm/gm/MapLabel.js

```diff
@@ -127,11 +127,45 @@ class MapLabel extends MapElement {
         ctx.lineWidth = strokeWeight;
         ctx.strokeText(text, x, y);
       }
+      this.wrapText(ctx, text, x, y);
+      // ctx.fillText(text, x, y);
+    }
+  }
 
-      ctx.fillText(text, x, y);
+  measureCanvas() {
+    var words = this.get('text').split('\r\n'),
+      metrics,
+      lineHeight,
+      y;
+    const context = this.canvas_.getContext('2d');
+    for (var i = 0; i < words.length; i++) {
+      var metrics = context.measureText(words[i]);
+      lineHeight =
+        1.2 *
+        (metrics.actualBoundingBoxAscent + metrics.actualBoundingBoxDescent);
+      if (metrics.width > this.canvas_.width) {
+        this.canvas_.width += metrics.width;
+      }
     }
+    if (lineHeight > this.canvas_.height) {
+      this.canvas_.height = y;
+    }
+    this.canvas_.width *= 2;
   }
 
+  wrapText(context, text, x, y) {
```

Comment:
Add comment block for method, including the typing of each arguments here and their documentation.